src/e2e-test/java/io/cdap/plugin/common/stepsdesign/GCS.java

```diff
@@ -0,0 +1,131 @@
+
+  public static void main(String[] args) {
+    validateOutputFileGeneratedByGCSSinkPluginIsEqualToExpectedOutputFile();
+  }
+
+  public static boolean validateJsonArray(JsonArray v1, JsonArray v2) {
+    if(!(v1.size() == v1.size())) return false;
+    for (int i =0 ; i < v1.size(); i++ ) {
+      if(!validateJsonElement(v1.get(i),v2.get(i))) return false;
+    }
+    return  true;
+  }
+
+  public static boolean validateJsonPrimitive(JsonElement v1, JsonElement v2) {
+    // TODO Update this later to a different function
+    return v1.getAsString().equals(v2.getAsString());
+  }
+
+  public static boolean validateJsonElement(JsonElement v1, JsonElement v2) {
+    // Taking v1 as source of truth
+    if (v1.isJsonObject()) {
+      // Base case of checking v2
+      if (!v2.isJsonObject()) return false;
+      return validateJsonObject(v1.getAsJsonObject(), v2.getAsJsonObject());
+
+    } else if (v1.isJsonArray()) {
+      if (!v2.isJsonArray()) return false;
+      return validateJsonArray(v1.getAsJsonArray(),v2.getAsJsonArray());
+
+    } else if (v1.isJsonPrimitive()) {
+      if (!v2.isJsonPrimitive()) return false;
+      return validateJsonPrimitive(v1, v2);
+
+    } else if (v1.isJsonNull()) {
+      if (!v2.isJsonNull()) return false;
+      return true;
+    }
+    return false;
+  }
+  public static boolean validateJsonObject(JsonObject v1, JsonObject v2) {
+    Set<Map.Entry<String, JsonElement>> v1EntrySet = v1.entrySet();
+    for (Map.Entry<String, JsonElement> entry : v1EntrySet) {
+      // Key
+      String key = entry.getKey();
+      // Values
+      JsonElement fileValue = v1.get(key);
+      JsonElement gcsValue = v2.get(key);
+      if (!validateJsonElement(fileValue, gcsValue)) return false;
+    }
+    return true;
+  }
+  public static void validateOutputFileGeneratedByGCSSinkPluginIsEqualToExpectedOutputFile() {
+    String projectId = "cdf-athena";
+    String bucketName = "cdf-e2e-test-473abc67-3b28-4f2a-8432-4e2f7619c57e";
+    String objectName = "2023-06-20-09-29/part-r-00000";
+    Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
+    BlobId blobId = BlobId.of(bucketName, objectName);
+    Blob blob = storage.get(blobId);
+    String[] gcsStrings = new String(blob.getContent()).split("\n");
+    String[] fileStrings = null;
+    try {
+      String filePath = "C:/Users/ishak/Downloads/google-cloud/src/e2e-test/resources/testdata/GCS/ExpectedOutputs/BQToDBAllDataTypes.json";
+      fileStrings = new String(Files.readAllBytes(Paths.get(filePath))).split("\n");
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    for (int i = 0; i <= gcsStrings.length - 1; i++) {
+      JsonObject gcsObject = gson.fromJson(gcsStrings[i], JsonObject.class);
+      JsonObject fileObject = gson.fromJson(fileStrings[i], JsonObject.class);
+
+      if (!validateJsonObject(fileObject, gcsObject)) {
+        System.out.println("FAILED");
```

Comment:
Instead of using print statement of getting the check FAILED/ PASSED. Please do use Assertion here.